### PR TITLE
[W5.11][W13-A3]Bay Wei Heng

### PR DESCRIPTION
**Pull request by bayweiheng**
Added three features, SortCommand, case insensitive commands, case insensitive FindCommand

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (aguss787) [inline src/seedu/addressbook/parser/Parser.java:14]: Don't use import `import seedu.addressbook.commands.*`
I think your IDE do this automatically?

File: doc/UserGuide.md [hunk 2]
```diff
@@ -53,16 +54,20 @@ Examples:
 Shows a list of all persons in the address book.<br>
 Format: `list`
 
+### Sorts all persons : `sort`
+Sorts all persons in the address book by name only. Ignores case. Relative order of persons with same name is preserved<br>
```
Comment (brandonyeoxg) [inline doc/UserGuide.md:58]: Perhaps you can elaborate if the name is sorted in ascending or descending order =)?

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package seedu.addressbook.commands;
+
+/**
+ * Clears the address book.
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/commands/SortCommand.java:4]: Perhaps this is more like `Sorts the names in the address book in ascending/descending order` =P?

File: src/seedu/addressbook/data/AddressBook.java [hunk 4]
```diff
@@ -107,6 +107,13 @@ public void clear() {
         allTags.clear();
     }
 
+    /**
+     * Stably sorts all persons in the address book by name.
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/AddressBook.java:111]: Same as previously stated, perhaps look to be more precise in stating the functionality of the method. Is it sorted is lexicographical order?

File: src/seedu/addressbook/commands/SortCommand.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    public static final String COMMAND_WORD = "sort";
+    public static final String MESSAGE_USAGE = "Sorts address book by name. Persons with the same name" +
+            "will have their relative order preserved. \n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "All persons in address book sorted!";
+
+    public SortCommand() {}
+
+
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/commands/SortCommand.java:17]: Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package seedu.addressbook.commands;
+
+/**
+ * Clears the address book.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+    public static final String MESSAGE_USAGE = "Sorts address book by name. Persons with the same name" +
+            "will have their relative order preserved. \n"
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/commands/SortCommand.java:10]: I think you got to add a space before `will` else it will appear weird on the help command =0.

**Review by brandonyeoxg — COMMENTED** [review]
@bayweiheng Some comments added. Please close the PR after reading comments. Great job keep it up!

File: src/seedu/addressbook/commands/SortCommand.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    public static final String MESSAGE_SUCCESS = "All persons in address book sorted!";
+
+    public SortCommand() {}
+
+
+    @Override
+    public CommandResult execute() {
+        addressBook.sortAllPersonsByName();
+        return new CommandResult(MESSAGE_SUCCESS);
```
Comment (Kyomian) [inline src/seedu/addressbook/commands/SortCommand.java:21]: So, after sorting, your program does not display the sorted address book?

**Pull request closed without merging**